site/docs/master/httpproxy.md

```diff
@@ -776,6 +776,37 @@ spec:
         replacement: /app
 ```
 
+### Header Policy
+
+HTTPProxy supports rewriting the `Host` header after first handling a request and before proxying to an upstream service.
+A common use-case for this is to use Contour to proxy to a resource outside the cluster referenced by an `externalName` service.
+
+The `headerRequestPolicy` supports a `replaceHeader` that currently only supports rewriting `Host` headers defined via a `key` and `value`.
+
+```yaml
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: header-rewrite-example
+spec:
+  virtualhost:
+    fqdn: header.bar.com
+  routes:
+  - services:
+    - name: s1
+      port: 80
+    headerRequestPolicy:
+      replaceHeader:
```

Comment:
qualifying these fields with `Header` feels redundant when seen in context like this.